### DISCUSSION TITLE
v4.0.0-beta2

### DISCUSSION THREAD

**Discussion by paulirwin** [paulirwin/JavaToCSharp#116] [Announcements]
## What's Changed
* GUI: Improve editor UI and add About dialog, #112 by @paulirwin in https://github.com/paulirwin/JavaToCSharp/pull/114
* Add support for file-scoped namespaces, #113 by @paulirwin in https://github.com/paulirwin/JavaToCSharp/pull/115

**Full Changelog**: https://github.com/paulirwin/JavaToCSharp/compare/v4.0.0-beta...v4.0.0-beta2

<hr /><em>This discussion was created from the release <a href='https://github.com/paulirwin/JavaToCSharp/releases/tag/v4.0.0-beta2'>v4.0.0-beta2</a>.</em>

**Comment by paulirwin** [comment]
@maximilien-noal I'd appreciate your assistance testing this release because of the Avalonia changes if you are able to!

**Reply by maximilien-noal** [reply]
Hello,

I'll glady do so. Right now all I can say is that it builds on my Xubuntu 23.10, and the GUI looks and behaves alright. I will test it with Java conversions later in the day.

**Reply by maximilien-noal** [reply]
Well actually I tested it just now, and here are the only bugs I found:

- When switching between dark and light mode several times, eventually code syntax colors are lost.
- The tooltips from the button on the top-right are cut off by the window. Internally, they should be popup windows (at least that's how it's done in WPF, IIRC) so this doesn't happen.

Maybe both are a limitation of the 3rd party theme used. Perhaps the 3rd party Material theme would be better ?

**Reply by paulirwin** [reply]
Thanks for looking into it! Regarding the tooltips, I wonder if that's a Linux issue. Doesn't seem to be a problem for me in macOS, actually I have the opposite problem: sometimes the tooltips remain on the screen when they shouldn't. I'm just using the regular Avalonia Tooltip attribute, so not sure what else I can do to fix that without an Avalonia fix. Let me know your thoughts. Just wanted to have a way of communicating to the user what the button is if the icon wasn't clear.

I haven't experienced the syntax coloring issue, but I'll try to repro that and see. If anything I think we might revert to using the Fluent theme because that's what AvaloniaEdit supports out of the box. I had to add the Fluent theme to the app styles and then override them with Semi.Avalonia in order for AvaloniaEdit to function properly, otherwise things like text selection was invisible/transparent.

**Reply by maximilien-noal** [reply]
I think both issues are fixed by the latest AvaloniaUI bugfix release.
